src/MergeContext.js

```diff
@@ -454,6 +454,201 @@ class BranchPosition
+                throw new Error(`bad character at line ${i}, offset ${invalidPosition}`);
+
+            if (line.length > 72)
+                throw new Error(`too long line '${line}'`);
+        }
+    }
+
+    // parses attributes of the raw PR description
+    _parse(trimmedPrDescription) {
+        const removedHeaderDescription = this._extractHeader(trimmedPrDescription);
+
+        try {
+            this._extractTrailer(removedHeaderDescription);
+        } catch (e) {
+            // TODO: supply debugString() prefix
+            Log.Logger.info("assuming no trailer: " + e.message);
+            this._body = removedHeaderDescription;
+        }
+
+        assert(this._body !== undefined && this._body !== null);
+
+        this._checkForTypos();
+    }
+
+    // author is a {name, value}
+    _parseAuthor(author) {
+        const cred = author.value.match(/^ ([\w][^@<>,]*) <(\S+@\S+\.\S+)>$/);
+        if (!cred)
+            throw new Error(`invalid '${author.name}' value format`);
+
+        if (cred[0].includes(','))
+            throw new Error(`invalid '${author.name}' value: commas are not allowed`);
+
+        return {name: cred[1].trim(), email: cred[2].trim()};
+    }
+
+    _extractHeader(trimmedPrDescription) {
+        if (trimmedPrDescription.startsWith('Authored-by:')) {
+            let tokenizer = new FieldsTokenizer(trimmedPrDescription);
+            const authorField = tokenizer.nextField();
+            this._author = this._parseAuthor(authorField);
+            if (!tokenizer.atEnd())
+                throw new Error(`header must have nothing but a single 'Authored-by' attribute`);
+            return this._trim(tokenizer.remaining());
+        } else {
+            return trimmedPrDescription;
+        }
+    }
+
+    _parseTrailer(trailer) {
+        let tokenizer = new FieldsTokenizer(trailer);
+
+        if (tokenizer.atEnd())
+            throw new Error(`an empty trailer`);
+
+        while (!tokenizer.atEnd()) {
+            const field = tokenizer.nextField();
+            if (field.name === "Co-authored-by")
+                this._parseAuthor(field);
+            // and skip any other field
+        }
+        // successfully parsed
+        this._trailer = trailer;
+    }
+
+    _extractTrailer(removedHeaderDescription) {
+        // index of the last occurrence of '\n\n'
+        const trailerIndex = removedHeaderDescription.search(/\n\n(?!.+\n\n)/s);
+
+        if (trailerIndex < 0) {
+            // check a special case when the entire message may represent a trailer
+            this._parseTrailer(removedHeaderDescription);
+            this._body = "";
+        } else {
+            const trailer = this._trim(removedHeaderDescription.substring(trailerIndex));
+            this._parseTrailer(trailer);
+            this._body = this._trim(removedHeaderDescription.substring(0, trailerIndex));
+        }
+    }
+
+    // checks whether the message has a misused attribute
+    _checkForTypos() {
```

Comment:
This method needs a `text` parameter. It should be used for both body and trailer parts.